### ISSUE TITLE
json error

### ISSUE THREAD

**Issue by istorepro** [tirupati17/currency-converter-swift3.0-viper#4]
i have this error 

Call can throw, but it is not marked with 'try' and the error is not handled
let jsonObject = JSON(data: response.data!)

`     Alamofire.request("http://api.fixer.io/latest?base=\(baseCurrencyCode)").responseJSON { response in
            print(response.request?.description)
            if response.result.isSuccess {
                let jsonObject = JSON(data: response.data!)
                completion?(jsonObject as AnyObject)
            } else {
                failed?(response.result.error as AnyObject)
            }
        }
    }
`

**Comment by tirupati17** [comment]
Hi @istorepro 

 Sorry for the delayed response.
  As per swift error handling introduction, we need to handle the error as produced by its expression. 
So there are basically two things we can do to handle it in our code for now. 

  1. `Do-Catch` - which handle the error by running a block of code 
2. `try?` - which handle it by converting it to an optional value.  

via `do-catch`  (Personally, prefer this for this situation because if something went wrong `catch` will call `failed()` closure):
```
    do {
        try let jsonObject = JSON(data: response.data!)
    }
    catch _ {
          failed?([]) //Just pass empty array for now  
    }
```

 via `try?`:    
```
     let jsonObject = try? JSON(data: response.data!)
```
 
Basically swift want you to do something with error as per error handling introduction https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/ErrorHandling.html#//apple_ref/doc/uid/TP40014097-CH42-ID508

Anyway, soon I'll update the current project to swift 4.0 compatible. 
Thanks!!
